lib/src/widgets/board.dart

```diff
@@ -729,13 +752,17 @@ class _BoardState extends State<Chessboard> {
       return;
     }
 
+    final square = widget.offsetSquare(details.localPosition);
```

Comment:
I think we want to put that after the test if current pointer down event is null. Because we want to ensure the tap up event is coming from the same pointer as the down event pointer.

And it made me think of something else. I think we're missing another check: we don't want to trigger the callback if the  square found on tap up is not the same as the one found on tap down. IE: the user taps a square and drags the finger to another square before releasing, in that case the onTappedSquare should not be called.

Reply:
I realise this makes the logic more complex, but can we avoid it?

Reply:
> I think we want to put that after the test if current pointer down event is null. Because we want to ensure the tap up event is coming from the same pointer as the down event pointer.

This doesn't work, because if the board is non-interactive and shape drawing is disabled, the pointerDown listener is disabled entirely, so current pointer down event will always be nullptr. Or should we just enable the pointerDown handler always as well?

Reply:
We probably want to enable all the `Listener` events all the time, otherwise it will be messy.

the `pointerDown` handler [already exit](https://github.com/lichess-org/flutter-chessground/blob/bc1bea6903749f2874928dc6d3cce3e83527ce63/lib/src/widgets/board.dart#L594) if the board is not interactive. So for the tapped square callback you want to put the logic before that line. Adding a new variable and register the square on tap down to compare it with the square on tap up should be enough.

Reply:
Done :+1: 

All tests still pass, and [these lines](https://github.com/lichess-org/flutter-chessground/pull/66/files#diff-6c1e160362ff42cf18a9ec4d30c845fff769793eb84d33e680ab9212e698293cR261-R265) verify that no we don't call `onTappedSquare` if pointer down and up are different squares (which is the case when dragging a piece)

Anything else that should be tested?

Reply:
I can add a test of course that tests the logic directly, just want to check to avoid redundant work

Reply:
I think a dedicated test would be good indeed. It adds another layer of documentation to understand the code and also we should test this in the case the board is not interactive (as opposed to during a piece drag where this callback is not really useful btw).

Reply:
Added a test :+1: